bitnami/postgresql/templates/_helpers.tpl

```diff
@@ -150,6 +150,96 @@ Return PostgreSQL port
 {{- end -}}
 {{- end -}}
 
+{{/*
+Return PostgreSQL master service type
+*/}}
+{{- define "postgresql.masterServiceType" -}}
+{{- if .Values.master.service -}}
+    {{- if .Values.master.service.type -}}
+      {{- .Values.master.service.type -}}
+    {{- else if .Values.service.type -}}
```

Comment:
What do you think about change this by:
```
{{- define "postgresql.masterServiceType" -}}
{{- serviceType := .Values.service.type }}
{{- if .Values.master.service -}}
    {{- if .Values.master.service.type -}}
      {{- serviceType = .Values.master.service.type -}}
    {{- end -}}
{{- end -}}
{{- serviceType -}}
{{- end -}}
```
From my point this is more readable.
The rest of functions can be rafactored in the same way.

Reply:
Yes great suggestion

Reply:
So after experimenting with our refactor suggestion I remembered why I had to settle on this solution in the first place. Because this logic also decides whether or not to include the value at all (i.e. Sometimes we include nodePort: and sometimes we don't, as in the case of clusterIP service type). When refactored in this way, it always assumes that we are returning a value. That is ok in the case of service type, but not for node port and clusterIP.

Reply:
Oh and even in the case of service type there is an issue. For example, if master.service.clusterIP is defined, then the check for .Values.master.service returns true but when it checks for .Values.master.service.type it will be false and then no service type is returned at all

Reply:
The control flow doesn't behave as might be expected and simply fall through to the original value

Reply:
Please let me know if I am missing something about how this templating helper function works...I am definitely new to using them

Reply:
Well, I am assuming we are setting `ClusterIP` as the default value to the `.Values.service.type` in the `values.yaml` file.

Reply:
ClusterIP doesn't have a default value, it is commented out by default